include/fakeit/StubbingProgress.hpp

```diff
@@ -425,11 +425,36 @@ namespace fakeit {
         struct ArgLocator {
             template<typename current_arg, typename ...T, int ...N>
             static void AssignArg(current_arg &&p, std::tuple<ArgValue<T, N>...> arg_vals) {
-                if (std::get<check_index>(arg_vals).pos == arg_index)
+#if __cplusplus >= 201703L
+                if constexpr (std::get<check_index>(arg_vals).pos == arg_index)
                     GetArg(std::forward<current_arg>(p)) = std::get<check_index>(arg_vals).value;
+#else
+                if (std::get<check_index>(arg_vals).pos == arg_index)
+                    Set(std::forward<current_arg>(p), std::get<check_index>(arg_vals).value);
+#endif
```

Comment:
Technically speaking `if constexpr` is just a syntactic sugar for template stuff, so it should be possible to have it as well in pre-C++17 I guess.

```cpp
void post_cpp17()
{
    if constexpr (true)
        {/*stuff*/}
    else
        {/*other stuff*/}
}

template <bool B>
void ifConstexprCompat() {/*stuff*/}

template <>
void ifConstexprCompat<false>() {/*other stuff*/}

void pre_cpp17()
{
    ifConstexprCompat<true>();
}
```

It would remove the need for exceptions.

Reply:
Yeah, I said it should be probably possible, but I'm not really up for it. Specially since std::get is not constexpr in C++11, it should need some work in that area too. 

BTW, edits are allowed so if somebody is willing to fix it, it's completely OK. I might also go for it, but probably takes some time.

Reply:
> std::get is not constexpr in C++11

Didn't knew, it's a bit annoying then I agree.